### PR TITLE
[Merged by Bors] - feat: port Data.Subtype

### PR DESCRIPTION
**Pull request by pechersky**

### CODE REVIEW

**Comment by pechersky** [comment]
I'd like help with the simplifier lints here -- 
- why is `val_eq_coe` proccing the linter? it's not tagged with `simp` anymore
- making `example`s of the LHS not simplifying, and trying to prove them using `simp` works, so I'm not sure what the problem is there

**Comment by digama0** [comment]
`val_eq_coe` is a syntactic tautology, as the linter says. That linter does not care about simp lemmas, the lemma itself says nothing. You should just remove it

File: Mathlib/Data/Subtype.lean [hunk 1]
```diff
@@ -3,175 +3,222 @@ Copyright (c) 2017 Johannes Hölzl. All rights reserved.
...
+-/
+
 
 open Function
 
 namespace Subtype
-variable {α : Sort _} {β : Sort _} {γ : Sort _} {p : α → Prop} {q : α → Prop}
 
-def simps.coe (x: Subtype p) : α := x
+#align subtype Subtype
```
Comment (semorrison) [inline Mathlib/Data/Subtype.lean:29]: This is unnecessary, I think. As far as I can see in the mathlib3port output, it's already happening correctly?

**Comment by semorrison** [comment]
bors merge

**Pull request closed without merging**